File: mobile/src/main/java/info/papdt/express/helper/model/BaseMessage.java
package info.papdt.express.helper.model;

public class BaseMessage<T> {

	private int code;
	private T object;

	public final static int CODE_OKAY = 200, CODE_ERROR = -1;

	public BaseMessage(int code, T object) {
		this.code = code;
		this.object = object;
	}

	public BaseMessage(int code) {
		this.code = code;
	}

	public BaseMessage() {

	}

	public int getCode() {
		return this.code;
	}

	public T getData() {
		return this.object;
	}

	public void setCode(int code) {
		this.code = code;
	}

	public void setData(T object) {
		this.object = object;
	}

}

File: mobile/src/main/kotlin/info/papdt/express/helper/model/BaseMessage.kt
package info.papdt.express.helper.model

class BaseMessage<T> {

    var code: Int = 0
    var data: T? = null

    constructor(code: Int, `object`: T) {
        this.code = code
        this.data = `object`
    }

    constructor(code: Int) {
        this.code = code
    }

    constructor()

    companion object {

        const val CODE_OKAY = 200
        const val CODE_ERROR = -1

    }

}
